java/arcs/core/storage/WriteBack.kt

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+package arcs.core.storage
+
+import androidx.annotation.VisibleForTesting
+import arcs.core.storage.keys.Protocols
+import java.util.concurrent.ExecutorService
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.asCoroutineDispatcher
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.sync.Mutex
+import kotlinx.coroutines.sync.withLock
+
+/**
+ * A layer to decouple local data updates and underlying storage layers flushes.
```

Comment:
How often/when do we write back today? Where do we explicitly request to persist? Are there any implications on how quickly/often particles are notified of handle data input changes prior to this PR? Does this revision have any implications for other storage features such as queries or streams? @csilvestrini @Cypher1

Reply:
The write-back is executed immediately by the initial implementation.
I don't want to defer write-back at this moment and don't implement other fancy features yet like merging, prioritizing, etc.
All I did is using dedicated write-back threads to do the write jobs instead of blocking current thread as what modern storage stack designs did.

As for how quickly particles are notified of handle data input changes prior to this PR?
The rough number I got is around 55-60 ms prior to the PR, around 25 ms with the PR.